npm/deploy/Deployer.kt

```diff
@@ -21,26 +21,49 @@
 
 package com.vaticle.bazel.distribution.npm.deploy
 
-import com.vaticle.bazel.distribution.common.Logging.Logger
 import com.vaticle.bazel.distribution.common.Logging.LogLevel.DEBUG
-import com.vaticle.bazel.distribution.common.OS.MAC
+import com.vaticle.bazel.distribution.common.Logging.Logger
 import com.vaticle.bazel.distribution.common.OS.LINUX
+import com.vaticle.bazel.distribution.common.OS.MAC
 import com.vaticle.bazel.distribution.common.shell.Shell
 import com.vaticle.bazel.distribution.common.shell.Shell.Command.Companion.arg
 import com.vaticle.bazel.distribution.common.util.SystemUtil.currentOS
+import com.vaticle.bazel.distribution.npm.deploy.Options.Env.DEPLOY_NPM_PASSWORD
+import com.vaticle.bazel.distribution.npm.deploy.Options.Env.DEPLOY_NPM_TOKEN
+import com.vaticle.bazel.distribution.npm.deploy.Options.Env.DEPLOY_NPM_USERNAME
 import java.nio.file.Files
 import java.nio.file.Path
+import java.util.*
 
 class Deployer(private val options: Options) {
     private val logger = Logger(logLevel = DEBUG)
 
     fun deploy() {
         Shell(logger = logger, verbose = true).execute(
-            command = Shell.Command(
-                arg("npm"), arg("publish"), arg("--registry=${options.registryURL}"),
-                arg("--${authParamFormattedURL(options.registryURL)}/:_authToken=${options.npmToken}", printable = false),
-                arg("deploy_npm.tgz")),
-            env = mapOf("PATH" to pathEnv()))
+                command = Shell.Command(
+                        arg("npm"), arg("publish"), arg("--registry=${options.registryURL}"),
+                        arg(authURI(options), printable = false),
+                        arg("deploy_npm.tgz")),
+                env = mapOf("PATH" to pathEnv()))
+    }
+
+    private fun authURI(options: Options): String {
+        val token = options.npmToken;
+        val user = options.npmUsername
+        val pass = options.npmPassword
+        val uriPrefix = "--" + authParamFormattedURI(options.registryURL)
+        if (token != null) {
```

Comment:
If the token is there, username/pass cannot be provided, and vice versa. This is to make sure the usage is as unambiguous as possible.